ATS_API/Scripts/Save/ModdedSaveManagerService.cs

```diff
@@ -218,9 +224,9 @@ private string CleanGuid(string guid)
             .Replace("|", "").Replace(".", "_");
     }
     
-    [HarmonyPatch(typeof(AppServices), nameof(AppServices.CreateServices))]
+    [HarmonyPatch(typeof(MetaServices), nameof(MetaServices.CreateServices))]
     [HarmonyPostfix]
-    private static void AddModdedSaveManagerAsService(AppServices __instance)
+    private static void AddModdedSaveManagerAsService(MetaServices __instance)
```

Comment:
Does this mean all loaded mods will load their save data when starting or loading a settlement?
So no save data will exist when in the world map for example?

Reply:
No, 
the acutual process is here:

## Launch the game

- `MainController.Awake()`
- `MainController.InitServices()`
  - 1: create `AppServices` (also create `ProfileServices`)
  - 2: `await AppServices.Load();`
    - This will Load `ProfileServices`
      - In `ProfileServices`, load `Profiles.save`, and setup the current profile `ProfilesService.Current`

- `MetaController.Awake()`
- `MainController.InitServices()`
  - Create our save service, and load it.

(It seems `MetaController` Awake after `MainController`)

## Switch Profiles

Create and Remove profiles: Invoke relevant methods in `ProfilesService`
 - Invoke `public async UniTask ProfilesService.ChangeProfile(ProfileData profile)`
  - Invoke `await this.PerformLogout(profile, null);` (` ClearProfile(ProfileData profile` will also invoke this)
    - In this method, it 
```C#
private async UniTask PerformLogout(ProfileData newProfile, Action<ProfileData> transitionCallback = null)
{
	Log.Info("[PROF] Perform logout", null);
	Serviceable.MetaController.DestroyController(); // they remove the meta controller gameobject
	await Serviceable.Controller.LoadTransition();
	if (transitionCallback != null)
	{
		transitionCallback(this.Current);
	}
	newProfile.lastPickTime = Serviceable.TimeService.Time;
	this.Current = newProfile; // switch to the target current profile
	await Serviceable.Controller.LoadMenu(); //Load scene 0, the scene loads the meta controller gameobject
}
```

The code will let the save & load behavior depends on the life cycle of `MetaController` `MetaServices`.
`MetaController` is kept as a singleton across scene by `EnforceSingleton()`, when it `Awake()`, if it finds it is not singleton, it will not execute the initialize code.

`IMetaController.DestroyController()` is only invoked by `ProfilesService.PerformLogout(...)`
So this is the only possible case that remove `MetaServices`.

I think API needs to handle `Profiles` concept anyway, otherwise as a modder I need to handle it by myself. (or just ignore it and hope no players open a new profile or start a new QHT run).

If we need a save data unrelated to profiles, then I need to write another series of code. (Example: Profiles data, setting data across all profiles)

Reply:
Makes sense!
Sounds great!